postgres/datadog_checks/postgres/connections.py

```diff
@@ -65,9 +99,34 @@ def get_connection(self, dbname, ttl_ms):
                 db.rollback()
 
             deadline = datetime.datetime.now() + datetime.timedelta(milliseconds=ttl_ms)
-            self._conns[dbname] = ConnectionWithTTL(db, deadline)
+            self._conns[dbname] = ConnectionInfo(
+                connection=db,
+                deadline=deadline,
+                active=True,
+                last_accessed=datetime.datetime.now(),
+                thread=threading.current_thread(),
+            )
             return db
 
+    @contextlib.contextmanager
+    def get_connection_cm(self, dbname: str, ttl_ms: int, timeout: int = None) -> psycopg2.extensions.connection:
```

Comment:
Are you planning to update the current usages of `get_connection`? I see that the signature was changed so I'm not sure what the phases are. The TODO seems to refer to the broader.

If updating the current usages, this function with the suffix seems unnecessary. You should be able to make this the default behavior for the `get_connection` function